src/Plugin.Maui.ScreenRecording/ScreenRecordingService.android.cs

```diff
@@ -2,68 +2,137 @@
+        if (OperatingSystem.IsAndroidVersionAtLeast(26))
+        {
+            CreateNotificationChannel();
+
+            var contentTitle = intent?.GetStringExtra(ExtraContentTitle) ?? ScreenRecordingOptions.defaultAndroidNotificationTitle;
+            var contentText = intent?.GetStringExtra(ExtraContentText) ?? ScreenRecordingOptions.defaultAndroidNotificationText;
+
+            var notification = new Notification.Builder(this, ChannelId)
+                .SetContentTitle(contentTitle)
+                .SetContentText(contentText)
+                .SetSmallIcon(global::Android.Resource.Drawable.PresenceVideoOnline)
+                .Build();
+
+            // Android Q
+            if (OperatingSystem.IsAndroidVersionAtLeast(29))
+            {
+                StartForeground(NotificationId, notification, ForegroundService.TypeMediaProjection);
+            }
+            else
+            {
+                StartForeground(NotificationId, notification);
+            }
+        }
+        else
+        {
+            // Pre-Android O
+            StartForeground(NotificationId, CreateFallbackNotification());
+        }
+    }
+
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Interoperability", "CA1416:Validate platform compatibility", Justification = "The callee bears responsibility for ensuring an OS version check before invoking this method.")]
+    private void CreateNotificationChannel()
+    {
+        var channel = new NotificationChannel(ChannelId, "Screen Recording Service", NotificationImportance.Default)
+        {
+            Description = "Notification Channel for Screen Recording Service"
+        };
+
+        var notificationManager = (NotificationManager?)GetSystemService(NotificationService);
+        notificationManager?.CreateNotificationChannel(channel);
+    }
+
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Interoperability", "CA1422:Validate platform compatibility", Justification = "The callee bears responsibility for ensuring an OS version check before invoking this method.")]
+    private Notification CreateFallbackNotification()
+    {
+        return new Notification.Builder(this)
+            .SetContentTitle("Screen Recording")
+            .SetContentText("Screen recording is running.")
+            .SetSmallIcon(global::Android.Resource.Drawable.PresenceVideoOnline)
+            .Build();
+    }
+
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Interoperability", "CA1422:Validate platform compatibility", Justification = "Compiler noob")]
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Interoperability", "CA1416:Validate platform compatibility", Justification = "Compiler noob")]
+    private static T? GetParcelableExtra<T>(Intent? intent, string name) where T : Java.Lang.Object =>
+        OperatingSystem.IsAndroidVersionAtLeast(33)
+            ? intent?.GetParcelableExtra(name) as T
+            : intent?.GetParcelableExtra(name, Java.Lang.Class.FromType(typeof(T))) as T;
```

Comment:
```suggestion
        OperatingSystem.IsAndroidVersionAtLeast(33)
            ? intent?.GetParcelableExtra(name, Java.Lang.Class.FromType(typeof(T))) as T
            :  intent?.GetParcelableExtra(name) as T;
```

I think the OS version check is wrong, if I swap the functions around here it does work.

Reply:
Oops 😅